docs/source/howto/workflows.rst

```diff
@@ -20,6 +20,159 @@ Running a predefined workflow
 
 `#3992`_
 
+The first step to running a predefined workflow is loading the work function or work chain class that defines the workflow you want to run.
+This can of course be achieved by importing the function or class from its respective module:
+
+.. code-block:: python
+
+    from aiida.workflows.plugins.arithmetic.addmultiply import add_and_multiply              # Import the addmultiply work function
+    from aiida.workflows.plugins.arithmetic.multiplyadd import MultiplyAndAddWorkChain       # Import the MultiplyAndAddWorkChain WorkChain
+
+However, a more robust way of loading a ``WorkChain`` is using the ``WorkflowFactory``, for example:
+
+.. code-block:: python
+
+    MultiplyAndAddWorkChain = WorkflowFactory('arithmetic.multiplyadd')
+
+The input argument of the ``WorkflowFactory`` is a string called the *entry point*.
+For each work chain, the entry point usually starts with with the name of the plugin package (``arithmetic``), followed by a dot and the string identifying the work chain (``multiplyadd``).
+Using the ``WorkflowFactory`` has the advantage that the entry point will not change when the packages or plugins are reorganised, and hence your code is less likely to break when updating AiiDA or the plugin that supplies the work chain.
```

Comment:
Naive question: why is using a "factory" liked this better than ensuring that our import paths remain stable.

Reply:
Because the same content also applies to other plugins and not just `aiida-core`, so it is a good paradigm to get used to. In addition, the entry point names are unique identifiers that are supported in other parts of the API. For example, in CLI commands such as `verdi plugin list` and many other. Also in the Python API there are places where these strings come back. This is not the case with the module path. So I think for users in general it is very useful to start thinking about the entry point names, instead of the full module path of the resource

Reply:
@sphuber Your explanation had the missing piece about the entry points from plugins so I think we should better motivate that in the docs.

Reply:
> @sphuber Your explanation had the missing piece about the entry points from plugins so I think we should better motivate that in the docs.

Should I add another sentence or two in the note at the end of this section, e.g.:

"Using the ``WorkflowFactory`` is a good paradigm to get used to as the entry point names are unique identifiers that are supported in many parts of AiiDA's API, such as the `verdi plugin list` command shown above."